SteamKit2/SteamKit2/Util/StreamHelpers.cs

```diff
@@ -64,28 +63,31 @@ public static float ReadFloat( this Stream stream )
             return BitConverter.ToSingle( data );
         }
 
+        const string NullTerminator = "\0";
+
         public static string ReadNullTermString( this Stream stream, Encoding encoding )
         {
             if ( encoding == Encoding.UTF8 )
             {
                 return ReadNullTermUtf8String( stream );
             }
 
-            int characterSize = encoding.GetByteCount( "e" );
+            using MemoryStream ms = new MemoryStream( capacity: 32 );
 
-            using MemoryStream ms = new MemoryStream();
+            int characterSize = encoding.GetByteCount( "e" );
+            Span<byte> data = stackalloc byte[ characterSize ];
 
             while ( true )
             {
-                byte[] data = new byte[ characterSize ];
-                stream.Read( data, 0, characterSize );
+                data.Clear();
+                stream.Read( data );
 
-                if ( encoding.GetString( data, 0, characterSize ) == "\0" )
+                if ( encoding.GetString( data ) == NullTerminator )

```

Comment:
Does this allocate a new string every time?

Reply:
Yeah. As I said above though, all the paths use UTF8 now so this won't even be hit.